app/static/js/_navbar.js

```diff
@@ -10,10 +10,74 @@ const minEditZoom = 13
 const navbar = document.querySelector(".navbar")
 const editGroup = navbar.querySelector(".edit-group")
 const loginLinks = navbar.querySelectorAll("a[href='/login']")
-
-// Configure the remote edit button (JOSM)
 const remoteEditButton = navbar.querySelector(".remote-edit")
-if (remoteEditButton) configureRemoteEditButton(remoteEditButton)
+
+const editButtons = editGroup.querySelectorAll(".dropdown-item.edit-link")
+const rememberChoice = editGroup.querySelector("input[name='remember-choice']")
+
+if (rememberChoice.disabled) {
+    Tooltip.getOrCreateInstance(rememberChoice.parentElement, {
+        title: rememberChoice.parentElement.dataset.bsTitle,
+        placement: "bottom"
+    }).enable();
+}
+
+const prepareEdit = (event) => {
+    const editButtonClicked = event.currentTarget;
+
+    if (!rememberChoice || !rememberChoice.checked) {
+        Dropdown.getInstance(
+            editGroup.querySelector("button.dropdown-toggle")
+        ).hide();
+
+        if (editButtonClicked.dataset.osmEditor == "remote") {
+            prepareRemoteEdit(editButtonClicked);
+        }
+        return;
+    };
+
+    // Set default editor when "remember my choice" is checked
+    event.preventDefault();
+    console.debug("Changing default editor to", editButtonClicked.dataset.osmEditor);
+
+    const defaultEditorBadge = editGroup.querySelector("span.badge.default-editor");
+    defaultEditorBadge.remove();
+    defaultEditorBadge.classList.replace("bg-green", "bg-secondary");
+    editButtonClicked.insertAdjacentElement("beforeend", defaultEditorBadge);
+
+    const userEditor = new FormData();
+    userEditor.append("editor", editButtonClicked.dataset.osmEditor);
+    fetch("/api/web/user/settings/editor", {
+			method: "POST",
+			body: userEditor,
+            mode: "same-origin",
+            cache: "no-store",
+            priority: "high",
+		}).then((response) => {
+            if (!response.ok) {
+                throw new Error(`${response.status} ${response.statusText}`);
+            }
+
+            console.debug("Changed default editor to", editButtonClicked.dataset.osmEditor);
+            defaultEditorBadge.classList.replace("bg-secondary", "bg-green")
+            uncheckRememberChoice();
+
+			editButtonClicked.dispatchEvent(new MouseEvent("click"));
```

Comment:
Do we need a new MouseEvent? Perhaps we could reuse the original:

```suggestion
			editButtonClicked.dispatchEvent(event);
```

Reply:
We need the new `MouseEvent`, since reusing the original event prevents iD and Rapid from being opened.

Reply:
Makes sense (because of preventDefault).

The event often holds useful information and I [searched a way](https://stackoverflow.com/a/20541207) to easily duplicate it. I have updated the suggested change above :slightly_smiling_face: